internal/runtime/client/client.go

```diff
@@ -302,3 +412,13 @@ func urlForExtension(config runtimev1.ClientConfig, gvh catalog.GroupVersionHook
 	u.Path = path.Join(u.Path, catalog.GVHToPath(gvh, name))
 	return u, nil
 }
+
+type errCallingExtensionHandler struct {
+	extensionHandlerName string
+	err                  error
+}
+
+func (e *errCallingExtensionHandler) Error() string {
+	// FIXME: find a better error message.
+	return fmt.Sprintf("failed processing handler %s with error: %s", e.extensionHandlerName, e.err)
```

Comment:
What about
```suggestion
	return fmt.Sprintf("failed to call Extensionhandler %q: %v", e.extensionHandlerName, e.err)
```

Reply:
The same error will be returned for network error, non 200 response objects and response decoding errors. Not sure if "failed to call...." is the right wording for all of them.

Reply:
Then the name of the struct is not ideal :)

What about simplifying this. We can make this custom type just:
```
type errCallingExtensionHandler error
```
and then we can write the exact right error message when we return. I think making those errors detectable one layer above does not necessarily mean we need a common error message style, e.g.
```
return errCallingExtensionHandler(errors.Wrapf(err, "failed to call ExtensionHandler: %q", opts.name))
```
(or if we have to keep the struct because we need access to `extensionHandlerName` we can just return `e.err.Error()`).